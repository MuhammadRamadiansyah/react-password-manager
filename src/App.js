import React, { Component } from 'react';
import Home from './components/Home'
import { NavbarHeader } from './components/NavbarHeader'
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavbarHeader />
        {/* <header className="App-header"> */}
          {/* <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1> */}
        {/* </header> */}
        <Home />
      </div>
    );
  }
}

export default App;
