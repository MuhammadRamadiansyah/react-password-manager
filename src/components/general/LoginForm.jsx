import React, { Component } from 'react'
import './LoginForm.css'
import { inject } from 'mobx-react';
import UserStore from '../../stores/UserStore';

@inject('UserStore')
class LoginForm extends Component {
  constructor () {
    super()
    this.state = {
      email: '',
      password: ''
    }
  }
  closeModal () {
    var modal = document.getElementById('loginModal')
    modal.style.display = "none"
  }

  handleChangeEmail = (e) => {
    this.setState({
      email: e.target.value
    })
    document.getElementById("message").style.display = "none";
  }

  handleChangePassword =(e) => {
    this.setState({
      password: e.target.value
    })
  }

  clearForm () {
    this.setState({
      email: '',
      password: ''
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.closeModal()
    this.clearForm()
    UserStore.login(this.state.email, this.state.password)
  }

  render() {
    return (
      <div>
        <div className="modal-header">
          <h1> Login Form </h1>
        </div>
        <form>
          <div className="container">
            <label htmlFor="email"><b>Email</b></label>
            <input type="text" placeholder="Enter Email" onChange={ this.handleChangeEmail } name="email" required  value ={this.state.email}/>
            <label htmlFor="psw"><b>Password</b></label>
            <input type="password" placeholder="Enter Password" onChange={ this.handleChangePassword } name="password" required value ={this.state.password}/>
            <div className="container actionButton">
              <button type="button" onClick={ this.closeModal } className="cancelbtn">Cancel</button>
              <button type="submit" onClick= { this.handleSubmit } >Submit</button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default LoginForm