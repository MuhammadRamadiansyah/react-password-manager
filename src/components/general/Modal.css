.modal {
  display: none; /* Hidden by default */
  position: fixed; /* Stay in place */
  z-index: 1; /* Sit on top */
  padding-top: 100px; /* Location of the box */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
}

.modalbtn{
  margin: 0px 5px;
}
/* Modal Content */
.modal-content {
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 50%;
  border-radius: 12px;
}

/* The Close Button */
.close {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
}

.scale-up-center {
	-webkit-animation: scale-up-center 0.4s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
	        animation: scale-up-center 0.4s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
}

@-webkit-keyframes scale-up-center {
  0% {
    -webkit-transform: scale(0.5);
            transform: scale(0.5);
  }
  100% {
    -webkit-transform: scale(1);
            transform: scale(1);
  }
}
@keyframes scale-up-center {
  0% {
    -webkit-transform: scale(0.5);
            transform: scale(0.5);
  }
  100% {
    -webkit-transform: scale(1);
            transform: scale(1);
  }
}
