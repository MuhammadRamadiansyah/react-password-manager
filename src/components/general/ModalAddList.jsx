import React, { Component } from 'react'
import AddListForm from './AddListForm.jsx'
import './ModalAddList.css'

export default class Modal extends Component {

  componentDidMount() {
    var modal = document.getElementById('addListModal')
    window.onclick = function(event) {
      if (event.target === modal) {
        modal.style.display = "none";
      } 
    }
  }
  openModal = () => {
    // var modal = document.getElementById('addListModal')
    // modal.style.display = "block"
  }

  logout = () => {
    this.props.logout()
  }

  render () {
    return (
      <div>
        <button className="modalbtn" type="button" onClick={this.openModal}> Add List</button>
        <button type="button" 
                className="cancelbtn" 
                onClick={this.logout}
                style={{
                  margin: '0px 10px'
                }}> Logout </button> 
        <div id="addListModal" className="modal scale-up-center">
          <div className="modal-content">
            <AddListForm title="add"/>    
          </div>
        </div>
      </div>
    )
  }
}