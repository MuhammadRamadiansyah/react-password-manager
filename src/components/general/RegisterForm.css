input[type=text], input[type=password] {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
}

.actionButton {
  text-align: end;
  margin: 0px 5px;
}

#message {
  display:none;
  color: #000;
  position: relative;
  padding: 20px;
  margin-top: 10px;
  text-align: left;
}

#message p {
  padding: 10px 35px;
  font-size: 18px;
}

.valid {
  color: green;
}

.valid:before {
  position: relative;
  left: -35px;
  content: "✔";
}

.invalid {
  color: red;
}

.invalid:before {
  position: relative;
  left: -35px;
  content: "✖";
}