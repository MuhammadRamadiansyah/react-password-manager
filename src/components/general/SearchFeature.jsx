import React, { Component } from 'react'

class SearchFeature extends Component {
  constructor () {
    super()
    this.state = {
      search: ''
    }
  }

  handleSearch = (e) => {
    this.setState({
      search: e.target.value
    })
  }
  handleSubmit = (e) => {
    this.props.submit(e, this.state.search)
  }

  render() {
    return (
      <div>
        <form>
          <input 
            type="text" 
            onChange ={this.handleSearch} 
            value= {this.search} 
            style = {{
              width: '40%',
              margin: '0px 10px',
            }}/>
            <button type="button" onClick = { this.handleSubmit }> Search </button>
        </form>
      </div>
    );
  }
}

export default SearchFeature