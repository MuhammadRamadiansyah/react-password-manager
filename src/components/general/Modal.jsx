import React, { Component } from 'react'
import './Modal.css'
import RegisterForm from './RegisterForm'
import LoginForm from './LoginForm'
import { inject } from 'mobx-react';

inject('UserStore')
export default class Modal extends Component {

  // componentDidMount() {
  //   var modal = document.getElementById('myModal')
  //   var loginModal = document.getElementById('loginModal')
  //   window.onclick = function(event) {
  //     if (event.target === modal) {
  //       // modal.style.display = "none";
  //     } else if (event.target === loginModal) {
  //       // loginModal.style.display = "none";
  //     }
  //   }
  // }
  openModal = () => {
    // var modal = document.getElementById('myModal')
    // modal.style.display = "block"
  }

  openLoginModal = () => {
    var modal = document.getElementById('loginModal')
    // modal.style.display = "block"
  }

  render () {
    return (
      <div>
        <button className="modalbtn" id="openRegister" type="button" onClick={this.openModal}>Register</button>
        <button className="modalbtn" id="openLogin" type="button" onClick={this.openLoginModal}>Login</button>
        <div id="myModal" className="modal scale-up-center">
          <div className="modal-content">
            <RegisterForm />    
          </div>
        </div>
        <div id="loginModal" className="modal scale-up-center">
          <div className="modal-content">
            <LoginForm />
          </div>
        </div>
      </div>
    )
  }
}