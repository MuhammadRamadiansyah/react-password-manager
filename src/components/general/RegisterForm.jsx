import React, { Component } from 'react'
import './RegisterForm.css'

class RegisterForm extends Component {
  constructor () {
    super()
    this.state = {
      email: '',
      password: ''
    }
  }
  closeModal () {
    var modal = document.getElementById('myModal')
    modal.style.display = "none"
  }

  handleChangeEmail = (e) => {
    this.setState({
      email: e.target.value
    })
    document.getElementById("message").style.display = "none";
  }

  handleChangePassword =(e) => {
    this.setState({
      password: e.target.value
    })
    document.getElementById("message").style.display = "block";

    let letter = document.getElementById("letter");
    let capital = document.getElementById("capital");
    let number = document.getElementById("number");
    let length = document.getElementById("length")

    let lowerCaseLetters = /[a-z]/g;
    if(this.state.password.match(lowerCaseLetters)) {  
      letter.classList.remove("invalid");
      letter.classList.add("valid");
    } else {
      letter.classList.remove("valid");
      letter.classList.add("invalid");
    }
    
    // Validate capital letters
    let upperCaseLetters = /[A-Z]/g;
    if(this.state.password.match(upperCaseLetters)) {  
      capital.classList.remove("invalid");
      capital.classList.add("valid");
    } else {
      capital.classList.remove("valid");
      capital.classList.add("invalid");
    }

    // Validate numbers
    let numbers = /[0-9]/g;
    if(this.state.password.match(numbers)) {  
      number.classList.remove("invalid");
      number.classList.add("valid");
    } else {
      number.classList.remove("valid");
      number.classList.add("invalid");
    }
    
    // Validate length
    if(this.state.password.length >= 8) {
      length.classList.remove("invalid");
      length.classList.add("valid");
    } else {
      length.classList.remove("valid");
      length.classList.add("invalid");
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.closeModal()
  }

  render() {
    return (
      <div>
        <form>
          <div className="container">
            <label htmlFor="email"><b>Email</b></label>
            <input type="text" placeholder="Enter Email" onChange={ this.handleChangeEmail } name="email" required  value ={this.state.email}/>

            <label htmlFor="psw"><b>Password</b></label>
            <input type="password" placeholder="Enter Password" onChange={ this.handleChangePassword } name="password" required value ={this.state.password}/>
              
            <div className="container actionButton">
              <button type="button" onClick={ this.closeModal } className="cancelbtn">Cancel</button>
              <button type="submit" onClick= { this.handleSubmit } >Submit</button>
            </div>
          </div>
        </form>
        <div id="message">
          <h3>Password must contain the following:</h3>
          <p id="letter" className="invalid">A <b>lowercase</b> letter</p>
          <p id="capital" className="invalid">A <b>capital (uppercase)</b> letter</p>
          <p id="number" className="invalid">A <b>number</b></p>
          <p id="length" className="invalid">Minimum <b>8 characters</b></p>
        </div>
      </div>
    );
  }
}

export default RegisterForm