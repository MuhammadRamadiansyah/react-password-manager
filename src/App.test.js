import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App from './App'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import NavbarHeader from './components/NavbarHeader'
import Home from './components/Home'
import UserStore from './stores/UserStore'
import { Provider } from 'mobx-react'
import Modal from './components/general/Modal'
import ModalAddList from './components/general/ModalAddList'
import SearchFeature from './components/general/SearchFeature'
import RegisterForm from "./components/general/RegisterForm"
import LoginForm from "./components/general/LoginForm"
import TableManager from "./components/homes/TableManager"

// import { wrap } from 'module'
import localStorageMock from './LocalStorageMock'
Enzyme.configure({ adapter: new Adapter() })

describe('<App /> rendering', () => {
  it('should render navbar header component and home', () => {
    const wrapper = shallow(<Provider UserStore={UserStore}><App /></Provider>)
    expect(wrapper.containsAllMatchingElements([
      <div />,
      <NavbarHeader />,
      <Home />
    ]))
  })
})

describe('<NavbarHeader /> render', () => {
  
  it('should has component <Modal /> when it is not login', () => {
    UserStore.isLogin = false
    const wrapper = shallow(<NavbarHeader />)
    expect(wrapper.containsAllMatchingElements([
      <Modal />
    ]))
    expect(wrapper.containsMatchingElement(<ModalAddList />)).toBeFalsy()
    expect(wrapper.containsMatchingElement(<SearchFeature />)).toBeFalsy()
  })

  it('should has component <ModalAddList /> and <SeacrhFeature />', () => {
    UserStore.isLogin = true
    const wrapper = shallow(<NavbarHeader />)
    expect(wrapper.containsAllMatchingElements([
      <ModalAddList />,
      <SearchFeature />
    ]))
    expect(wrapper.containsMatchingElement(<Modal />)).toBeFalsy()
  })
})

describe('<Modal /> render', () => {
  it('should has component <RegisterForm /> and <LoginForm />', () => {
    const wrapper = shallow(<div />)
    expect(wrapper.containsAllMatchingElements([
      <LoginForm />,
      <RegisterForm />
    ]))
  })

  it('login and register modal should has style display none before its called and display block after its called', () => {
    const modal = shallow(<Modal />)
    const registerModal = shallow(<div id="myModal"/>)
    const loginModal = shallow(<div id="loginModal"/>)
    expect(registerModal.hasClass('scale-up-center'))
    expect(registerModal.hasClass('modal'))
    expect(loginModal.hasClass('modal'))
    expect(loginModal.hasClass('scale-up-center'))

    //Modal Button open works
    const instance = modal.instance()
    const openModal = jest.spyOn(instance, 'openModal')
    modal.instance().openModal()

    const openLoginModal = jest.spyOn(instance, 'openLoginModal')
    modal.instance().openLoginModal()

    expect(openModal).toBeCalled()
    expect(openLoginModal).toBeCalled()

  })
})

describe('Register and Login form test', () => {

  describe('Register form all button are works', () => {
    it('email handle change, password handle change, and submit are works', () => {
      const component = renderer.create(<RegisterForm />)
      const wrapper = shallow(<RegisterForm />)
      expect(wrapper.state('email')).toBe('')
      expect(wrapper.state('password')).toBe('')
  
      wrapper.find('#newEmail').simulate('change', 
      {target: {name: 'email', value:'rama@gmail.com'}})
      wrapper.find('#newPassword').simulate('change', 
      {target: {name: 'password', value:'Rama12345!'}})
  
      expect(wrapper.state('email')).toBe('rama@gmail.com')
      expect(wrapper.state('password')).toBe('Rama12345!')
    
      expect(component).toMatchSnapshot()
  
      //close modal button works
      const instance = wrapper.instance()
      const close = jest.spyOn(instance, 'closeModal')

      //clear modal button works
      const clear = jest.spyOn(instance, 'clearForm')

      //submit modal button works
      const submit = jest.spyOn(instance, 'handleSubmit')
      // wrapper.instance().handleSubmit({ preventDefault() {} })
      // expect(close).toHaveBeenCalledTimes(1)
      // expect(clear).toHaveBeenCalledTimes(1)
      // expect(submit).toBeCalled()

      // expect(UserStore.register(wrapper.state('email'), wrapper.state('password'))).toBeCalled()
      // expect(wrapper.state('email')).toBe('')
      // expect(wrapper.state('password')).toBe('')
    })
  })
})

describe('application list', () => {

})

describe('UserStore testing', () => {

  var loginEmail = ''
  var loginPassword = ''
  var strongRegex = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})")

  beforeEach (() => {
    const wrapper = shallow(<LoginForm />)
    expect(wrapper.state('email')).toBe('')
    expect(wrapper.state('password')).toBe('')

    wrapper.find('#loginEmail').simulate('change', 
    {target: {name: 'email', value:'rama2@gmail.com'}})
    wrapper.find('#loginPassword').simulate('change', 
    {target: {name: 'password', value:'Rama12345'}})

    loginEmail = wrapper.state('email')
    loginPassword = wrapper.state('password')
  })
  test('login successfull', async () => {
    await UserStore.login(loginEmail, loginPassword)
    expect(UserStore.userKey).toEqual(localStorageMock.getItem('userKey'))
    expect(UserStore.user.email).toEqual(loginEmail)
    UserStore.user.apps.forEach((element) => {
      expect(element).toHaveProperty('app')
      expect(element).toHaveProperty('email')
      expect(element).toHaveProperty('password')
      expect(element).toHaveProperty('updatedAt')
      expect(element).toHaveProperty('createdAt')
      expect(element).toHaveProperty('realPassword')
      expect(element.realPassword).toMatch(strongRegex)
    })
    expect(UserStore.isLogin).toBeTruthy()
  })

  test('generate table with using apps data from firebase', () => {
    const table = shallow(<TableManager UserStore={UserStore}/>)
    expect(table.props()).toHaveProperty('UserStore')
  })

  test('check logout feature', () => {
    const wrapper = shallow(<NavbarHeader />)
    const instance = wrapper.instance()
    const logout = jest.spyOn(instance, 'logout')
    expect(localStorageMock.getItem('userKey')).not.toBeNull()
    wrapper.instance().logout()
    expect(localStorageMock.getItem('userKey')).toBeNull()
  })
})