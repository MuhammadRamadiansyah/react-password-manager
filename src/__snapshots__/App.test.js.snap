// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render Home alert testing element 1`] = `
<div
  className="home-alert"
>
  <h1>
    How To Use This Apps?
  </h1>
  <p>
    1. Login at the right corner of the window
  </p>
  <p>
    2. If you don't have an account, please register yourself
  </p>
  <p>
    3. You can add your application url, account or email, and also password
  </p>
  <p>
    4. You can also edit and delete your application list
  </p>
  <p>
    5. If you have probles please contact administrator!
  </p>
</div>
`;

exports[`CRUD Testing, look Password, hide Password and search App create new app 1`] = `
<div>
  <div
    className="modal-header"
  >
    <h1>
       
      ADD
       LIST FORM
    </h1>
  </div>
  <form>
    <div
      className="container"
    >
      <label
        htmlFor="appName"
      >
        <b>
          App Name
        </b>
      </label>
      <input
        id="appName"
        name="app"
        onChange={[Function]}
        placeholder="Enter your apps link"
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        id="appEmail"
        name="email"
        onChange={[Function]}
        placeholder="Enter Email"
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="psw"
      >
        <b>
          Password
        </b>
      </label>
      <input
        id="appPassword"
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <div
        className="container actionButton"
      >
        <button
          className="cancelbtn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
  <div
    id="message"
  >
    <h3>
      Password must contain the following:
    </h3>
    <p
      className="invalid"
      id="letter"
    >
      A 
      <b>
        lowercase
      </b>
       letter
    </p>
    <p
      className="invalid"
      id="capital"
    >
      A 
      <b>
        capital (uppercase)
      </b>
       letter
    </p>
    <p
      className="invalid"
      id="number"
    >
      A 
      <b>
        number
      </b>
    </p>
    <p
      className="invalid"
      id="length"
    >
      Minimum 
      <b>
        8 characters
      </b>
    </p>
    <p
      className="invalid"
      id="special"
    >
      A 
      <b>
        special character
      </b>
    </p>
  </div>
</div>
`;

exports[`Register and Login form test Register form all button are works email handle change, password handle change, and submit are works 1`] = `
<div>
  <div
    className="modal-header"
  >
    <h1>
       Register Form 
    </h1>
  </div>
  <form>
    <div
      className="container"
    >
      <label
        htmlFor="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        id="newEmail"
        name="email"
        onChange={[Function]}
        placeholder="Enter Email"
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="psw"
      >
        <b>
          Password
        </b>
      </label>
      <input
        id="newPassword"
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <div
        className="container actionButton"
      >
        <button
          className="cancelbtn"
          id="cancelBtn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          id="submitBtn"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
  <div
    id="message"
  >
    <h3>
      Password must contain the following:
    </h3>
    <p
      className="invalid"
      id="letter"
    >
      A 
      <b>
        lowercase
      </b>
       letter
    </p>
    <p
      className="invalid"
      id="capital"
    >
      A 
      <b>
        capital (uppercase)
      </b>
       letter
    </p>
    <p
      className="invalid"
      id="number"
    >
      A 
      <b>
        number
      </b>
    </p>
    <p
      className="invalid"
      id="length"
    >
      Minimum 
      <b>
        8 characters
      </b>
    </p>
    <p
      className="invalid"
      id="special"
    >
      A 
      <b>
        special character
      </b>
    </p>
  </div>
</div>
`;

exports[`test each of components element <LoginForm /> component 1`] = `
<div>
  <div
    className="modal-header"
  >
    <h1>
       Login Form 
    </h1>
  </div>
  <form>
    <div
      className="container"
    >
      <label
        htmlFor="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        id="loginEmail"
        name="email"
        onChange={[Function]}
        placeholder="Enter Email"
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="psw"
      >
        <b>
          Password
        </b>
      </label>
      <input
        id="loginPassword"
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <div
        className="container actionButton"
      >
        <button
          className="cancelbtn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`test each of components element <ModalAddList /> component 1`] = `
<div>
  <button
    className="modalbtn"
    onClick={[Function]}
    type="button"
  >
     Add List
  </button>
  <button
    className="cancelbtn"
    onClick={[Function]}
    style={
      Object {
        "margin": "0px 10px",
      }
    }
    type="button"
  >
     Logout 
  </button>
  <div
    className="modal scale-up-center"
    id="addListModal"
  >
    <div
      className="modal-content"
    >
      <div>
        <div
          className="modal-header"
        >
          <h1>
             
            ADD
             LIST FORM
          </h1>
        </div>
        <form>
          <div
            className="container"
          >
            <label
              htmlFor="appName"
            >
              <b>
                App Name
              </b>
            </label>
            <input
              id="appName"
              name="app"
              onChange={[Function]}
              placeholder="Enter your apps link"
              required={true}
              type="text"
              value=""
            />
            <label
              htmlFor="email"
            >
              <b>
                Email
              </b>
            </label>
            <input
              id="appEmail"
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              required={true}
              type="text"
              value=""
            />
            <label
              htmlFor="psw"
            >
              <b>
                Password
              </b>
            </label>
            <input
              id="appPassword"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              required={true}
              type="password"
              value=""
            />
            <div
              className="container actionButton"
            >
              <button
                className="cancelbtn"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
        <div
          id="message"
        >
          <h3>
            Password must contain the following:
          </h3>
          <p
            className="invalid"
            id="letter"
          >
            A 
            <b>
              lowercase
            </b>
             letter
          </p>
          <p
            className="invalid"
            id="capital"
          >
            A 
            <b>
              capital (uppercase)
            </b>
             letter
          </p>
          <p
            className="invalid"
            id="number"
          >
            A 
            <b>
              number
            </b>
          </p>
          <p
            className="invalid"
            id="length"
          >
            Minimum 
            <b>
              8 characters
            </b>
          </p>
          <p
            className="invalid"
            id="special"
          >
            A 
            <b>
              special character
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
